tests/integration/serde/json.rs

```diff
@@ -0,0 +1,144 @@
+use std::error::Error;
+
+use serde::{Deserialize, Serialize};
+use serde_test::Configure;
+use time::{Weekday, Time, Month, PrimitiveDateTime, Date, OffsetDateTime, Duration};
+
+enum Format {
+    Compact,
+    Readable,
+}
+
+use Format::*;
+
+fn serialize<T: Serialize>(val: T, fmt: Format) -> Result<String, Box<dyn Error>> {
+    let mut buf: Vec<u8> = Vec::new();
+    let mut ser = serde_json::Serializer::new(&mut buf);
+    match fmt {
+        Compact => {
+            let ser = (&mut ser).compact();
+            val.serialize(ser)?;
+        }
+        Readable => {
+            let ser = (&mut ser).readable();
+            val.serialize(ser)?;
+        }
+    }
+
+    let str = String::from_utf8(buf)?;
+    Ok(str)
+}
+
+fn deserialize<'a, T: Deserialize<'a>>(from: &'a str, fmt: Format) -> Result<T, Box<dyn Error>> {
+    let mut de = serde_json::Deserializer::from_str(from);
+    let val = match fmt {
+        Compact => {
+            let de = (&mut de).compact();
+            T::deserialize(de)?
+        }
+        Readable => {
+            let de = (&mut de).readable();
+            T::deserialize(de)?
+        }
+    };
+    Ok(val)
+}
+
+#[test]
+fn weekday_json() -> Result<(), Box<dyn Error>> {
+    use Weekday::*;
+
+    assert_eq!(serialize(Monday, Compact)?, "1");
+    assert_eq!(deserialize::<Weekday>("1", Compact)?, Monday);
+
+    assert_eq!(serialize(Monday, Readable)?, "\"Monday\"");
+    assert_eq!(deserialize::<Weekday>("\"Monday\"", Readable)?, Monday);
+    assert_eq!(deserialize::<Weekday>("1", Readable)?, Monday);
+
+    Ok(())
+}
+
+#[test]
+fn month_json() -> Result<(), Box<dyn Error>> {
+    use Month::*;
+
+    assert_eq!(serialize(March, Compact)?, "3");
+    assert_eq!(deserialize::<Month>("3", Compact)?, March);
+
+    assert_eq!(serialize(March, Readable)?, "\"March\"");
+    assert_eq!(deserialize::<Month>("\"March\"", Readable)?, March);
+    assert_eq!(deserialize::<Month>("3", Readable)?, March);
+
+    Ok(())
+}
+
+#[test]
+fn time_json() -> Result<(), Box<dyn Error>> {
+    let time = Time::from_hms(12, 40, 20)?;
```

Comment:
For statically known values, it's best to use the macro.

Reply:
:+1: